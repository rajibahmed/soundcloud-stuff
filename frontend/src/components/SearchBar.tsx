import React from "react";

export default ({ setQuery }: { setQuery: Function }) => (
  <div className="SearchBarWrap">
    <form>
      <input
        type="text"
        name="song"
        onChange={(e) => e.target.value && setQuery(e.target.value)}
        className="SearchBar"
        placeholder="What is your favorite music?"
      />
    </form>
  </div>
);
