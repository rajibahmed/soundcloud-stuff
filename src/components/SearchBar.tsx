import React from "react";

type SearchBarProps = {
  setQuery: Function;
};

const SearchBar: React.FC<SearchBarProps> = ({ setQuery }) => (
  <div className="SearchBarWrap">
    <form>
      <input
        type="text"
        name="song"
        onChange={(e) => e.target.value && setQuery(e.target.value)}
        className="SearchBar"
        placeholder="What is your favorite music?"
      />
    </form>
  </div>
);

export default SearchBar;
