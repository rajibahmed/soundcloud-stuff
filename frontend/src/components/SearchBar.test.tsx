import React from "react";
import SearchBar from "./SearchBar";
import { render } from "@testing-library/react";

test("Link changes the class when hovered", () => {
  const noop = () => {};
  const component = render(<SearchBar setQuery={noop} />);

  expect(component.asFragment()).toMatchSnapshot();
});
