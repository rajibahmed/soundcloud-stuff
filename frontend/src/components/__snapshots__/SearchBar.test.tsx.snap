// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Default view 1`] = `
<DocumentFragment>
  <div
    class="SearchBarWrap"
  >
    <form>
      <input
        class="SearchBar"
        name="song"
        placeholder="What is your favorite music?"
        type="text"
      />
    </form>
  </div>
</DocumentFragment>
`;
